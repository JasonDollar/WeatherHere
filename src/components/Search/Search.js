import React from 'react'

import classes from './Search.module.scss'

const Search = ({ 
  onSearchFormSubmit, onInputChange, inputValue, text, showSearchHandler, getUserLocation,
}) => (
  <div className={classes.container}>
    <div className={classes.backdrop} />
    <div className={classes.Search}>
      <form onSubmit={onSearchFormSubmit} className={classes.form}>
        <input type="text" onChange={onInputChange} value={inputValue} className={classes.input} />
        <button type="submit" className={classes.button}>{text.search}</button>
      </form>
      <button onClick={getUserLocation}>Geo</button>
    </div>
  </div>
)

export default Search
