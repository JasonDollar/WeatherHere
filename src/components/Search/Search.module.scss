.container {
  flex: 1;
  overflow: hidden;
}

.Search {
  display: flex;
  flex-direction: column;
  align-content: flex-end;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 200;
  box-shadow: 2px 1px 10px -1px var(--color-box-shadow);
  border-radius: 7px;
  margin: 2.5rem 0.5rem;
  background-color: var(--color-grey-3);
}

.button {
  flex-direction: column;
  display: flex;
  justify-content: center;
  align-items: center;
  background: none;
  border: 1px solid transparent;
  cursor: pointer;
  & span {
    color: var(--color-grey-1);
  }
}

.form {
  flex: 8;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 1rem 1rem;
  position: relative;
  box-shadow: 0px 0px 10px -3px var(--color-box-shadow);

  & .input {
    line-height: normal;
    box-sizing: inherit;
    flex: 1;
    padding: 1rem 0 1rem 1.5rem;
    border: none;
    border-radius: 5px;
    font-size: var(--big-text-font-size);
    box-shadow: 2px 1px 3px -2px var(--color-box-shadow);

    background: var(--color-back2);
    outline: none;

    &:focus {
      border-bottom: 1px solid var(--color-text-secondary);
    }

    &::placeholder {
      color: var(--color-text-transparent);
    }
  }

  & .formButton {
    position: absolute;
    
    right: 1px;
    padding: .6rem .8rem .6rem 1rem;
    background: var(--color-back2);
    z-index: 110;
    border-left: 1px solid var(--color-text-secondary);
    font-size: 10px;

    & svg {
      height: var(--big-text-font-size);
      width: var(--big-text-font-size);
      fill: var(--color-primary-offset);
      transform: scale(1.2);
    }
  }
}


.buttonGeo {
  padding: 1rem 1rem;
  margin: 1rem auto;

  & svg {
    height: var(--big-text-font-size);
    width: var(--big-text-font-size);
    fill: var(--color-primary);
    transform: scale(1.5);
  }

  & span {
    padding-top: 1rem;
  }
}