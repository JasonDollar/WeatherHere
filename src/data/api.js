const DARK_API = 'a077c3a2c30623e41ef4a1cafbec622d'
const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_API, DARK_URL }  

