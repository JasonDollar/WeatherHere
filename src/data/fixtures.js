import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}