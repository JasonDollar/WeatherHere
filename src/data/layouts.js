const layouts = {
  pink: {
    id: 'pink',
    name: 'Pink',
    primary: '#dd0055',
    primaryOffset: '#f5689e',
    mobileMenu: '#dd0055',
    antiGraph: '#8884d8',
    secondaryGraph: '#0c1052',
    background: '#fffcfc',
    background2: '#fffcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#eae9e9',
    greyGraph: '#676767',
  },
  pinkDark: {
    id: 'pinkDark',
    name: 'Pink Dark',
    primary: '#dd0055',
    primaryOffset: '#dd0055',
    mobileMenu: '#f5689e',
    antiGraph: '#8884d8',
    secondaryGraph: '#dd0055',
    background: '#272424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#efeee7',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#333',
    greyGraph: '#acacac',
  },
  pumpkin: {
    id: 'pumpkin',
    name: 'Pumpkin Dark',
    primary: '#f48042',
    primaryOffset: '#ef915f',
    mobileMenu: '#f48042',
    antiGraph: '#50c25f',
    secondaryGraph: '#f48042',
    background: '#262424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eee',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  evening: {
    id: 'evening',
    name: 'Evening Dark',
    primary: '#b92b27',
    primaryOffset: '#d84f4b',
    mobileMenu: '#d84f4b',
    antiGraph: '#1565C0',
    secondaryGraph: '#d84f4b',
    background: '#282424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eeeef8',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  sky: {
    id: 'sky',
    name: 'Cool Sky',
    primary: '#2980B9',
    primaryOffset: '#6DD5FA',
    mobileMenu: '#2980B9',
    antiGraph: '#756ef0',
    secondaryGraph: '#2980B9',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7e7ec',
    greyGraph: '#676767',
  },
  harvey: {
    id: 'harvey',
    name: 'Harvey',
    primary: '#2b574b',
    primaryOffset: '#75d1a6',
    mobileMenu: '#386e60',
    antiGraph: '#0daeca',
    secondaryGraph: '#2b574b',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7eae7',
    greyGraph: '#676767',
  },


}

export default layouts



