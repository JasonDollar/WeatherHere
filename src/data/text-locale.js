const pl = {
  header: {
    title: 'Pogoda',
    search: 'Szukaj',
  },  
  layout: {
    search: 'Szukaj',
  },
  current: {
    today: 'Dziś',
    humidity: 'Wilgotność',
    wind: 'Wiatr',
    precipProb: 'Opady',
    precipType: 'Rodzaj opadu',
    precipInt: 'Intensywność opadu',
    appTemp: 'Temperatura odczuwalna',
    dewPoint: 'Punkt rosy',
    pressure: 'Ciśnienie',
    cloudCov: 'Zachmurzenie',
    visibility: 'Widoczność',
    uvIndex: 'Indeks UV',
    sunrise: 'Wschód słońca',
    sunset: 'Zachód słońca',
    moonphase: 'Faza księżyca',
    ozone: 'Ozon',
    windGust: 'Wiatr w porywach',
    location: 'Nieznana lokacja',
  },
  date: {
    weekDay: ['Niedziela', 'Poniedziałek', 'Wtorek', 'Środa', 'Czwartek', 'Piątek', 'Sobota', 'Dziś'],
    weekDayShort: ['Niedz.', 'Pon.', 'Wt.', 'Śr.', 'Czw.', 'Pt.', 'Sob.'],
    today: 'Dziś',
    temperature: 'Temperatura (°C)',
    temperatureMax: 'Temperatura Max.',
    temperatureMin: 'Temperatura Min.',
  },
  hourly: {
    hour: 'Godzina',
    precipProb: 'Możliwość Opadów (%)',
    precipProbShort: 'Możliwość Opadów',
    temperature: 'Temperatura (°C)',
    temperatureShort: 'Temperatura',
  },
  update: 'Ostatnia aktualizacja: ',
}

const en = {
  header: {
    title: 'Weather',
    
  },
  layout: {
    search: 'Search',
  },
  current: {
    today: 'Today',
    humidity: 'Humidity',
    wind: 'Wind',
    precipProb: 'Precipitation chance',
    precipType: 'Precipitation type',
    precipInt: 'Precipitation intensity',
    appTemp: 'Apparent temperature',
    dewPoint: 'Dew point',
    pressure: 'Pressure',
    cloudCov: 'Cloud cover',
    visibility: 'Visibility',
    uvIndex: 'UV Index',
    sunrise: 'Sunrise',
    sunset: 'Sunset',
    moonphase: 'Moon phase',
    ozone: 'Ozone',
    windGust: 'Wind gust',
    location: 'Unknown location',
  },
  date: {
    weekDay: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
    weekDayShort: ['Sun.', 'Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.'],
    today: 'Today',
    temperature: 'Temperature (°C)',
    temperatureMax: 'Temperature Max.',
    temperatureMin: 'Temperature Min',
  },
  hourly: {
    hour: 'Hour',
    precipProb: 'Precipitation chance (%)',
    precipProbShort: 'Precipitation chance',
    temperature: 'Temperature (°C)',
    temperatureShort: 'Temperature',
  },
  update: 'Last updated: ',
}

export { pl, en } 