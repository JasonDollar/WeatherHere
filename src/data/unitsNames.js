export const unitsNames = {
  si: {
    id: 'si',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'm/s',
    pressure: 'hPa',
  },
  us: {
    id: 'us',
    distance: 'mi',
    precip: 'in/h',
    precipAcc: 'in',
    temperature: '째F',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  uk2: {
    id: 'uk2',
    distance: 'mi',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  ca: {
    id: 'ca',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'km/h',
    pressure: 'hPa',
  },
}