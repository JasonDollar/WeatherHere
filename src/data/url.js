const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_URL }