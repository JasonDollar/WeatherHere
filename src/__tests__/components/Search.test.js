import React from 'react'
import { shallow } from 'enzyme'
import { appTestData as data } from '../../data/fixtures'
import Search from '../../components/Search/Search'

const testProps = {
  onSearchFormSubmit: jest.fn(),
  onInputChange: jest.fn(),
  inputValue: '',
  text: data.text.pl.layout,
  getUserLocation: jest.fn(),
  showBackdrop: true,
  hideBackdrop: jest.fn(),
  showSearch: true,
}
let wrapper

beforeAll(() => {
  wrapper = shallow(<Search {...testProps} />)
})

describe('Search component tests', () => {
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('simulate input change', () => {
    const payload = { target: 'City name' }
    const element = wrapper.find('input')
    element.simulate('change', payload)
    expect(testProps.onInputChange).toHaveBeenCalledTimes(1)
    expect(testProps.onInputChange).toHaveBeenLastCalledWith(payload)
  })

  test('geolocation button click', () => {
    const element = wrapper.find('.buttonGeo')
    element.simulate('click')
    expect(testProps.getUserLocation).toHaveBeenCalledTimes(1)
  })

  test('simulate form submition', () => {
    // const payload = { target: 'City name' }
    const element = wrapper.find('form')
    element.simulate('submit', { preventDefault: () => { } })
    expect(testProps.onSearchFormSubmit).toHaveBeenCalledTimes(1)
  })
})