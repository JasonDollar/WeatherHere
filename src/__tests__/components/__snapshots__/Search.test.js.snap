// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component tests Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    getUserLocation={[MockFunction]}
    hideBackdrop={[MockFunction]}
    inputValue=""
    onInputChange={[MockFunction]}
    onSearchFormSubmit={[MockFunction]}
    showBackdrop={true}
    showSearch={true}
    text={
      Object {
        "forecast": "Pogoda",
        "geo": "Geo",
        "geoFull": "Geolokalizacja",
        "icon": "Ikona",
        "inputPlaceholder": "Szukaj lokalizacji",
        "search": "Szukaj",
        "settings": "Ustawienia",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Backdrop
          hideBackdrop={[MockFunction]}
          showBackdrop={true}
        />,
        <div
          className="Search"
          style={
            Object {
              "opacity": "1",
              "transform": "translateY(0)",
            }
          }
        >
          <form
            className="form"
            onSubmit={[MockFunction]}
          >
            <input
              className="input"
              onChange={[MockFunction]}
              placeholder="Szukaj lokalizacji"
              type="text"
              value=""
            />
            <button
              className="button formButton"
              type="submit"
            >
              <Icon
                icon="search"
                title="icon"
                viewBox="0 0 50 50"
              />
              <span>
                Szukaj
              </span>
            </button>
          </form>
          <button
            className="button buttonGeo"
            onClick={[MockFunction]}
            type="button"
          >
            <Icon
              icon="gpsArrow"
              title="icon"
              viewBox="0 0 51.636 51.636"
            />
            <span>
              Geolokalizacja
            </span>
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "hideBackdrop": [MockFunction],
          "showBackdrop": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <form
              className="form"
              onSubmit={[MockFunction]}
            >
              <input
                className="input"
                onChange={[MockFunction]}
                placeholder="Szukaj lokalizacji"
                type="text"
                value=""
              />
              <button
                className="button formButton"
                type="submit"
              >
                <Icon
                  icon="search"
                  title="icon"
                  viewBox="0 0 50 50"
                />
                <span>
                  Szukaj
                </span>
              </button>
            </form>,
            <button
              className="button buttonGeo"
              onClick={[MockFunction]}
              type="button"
            >
              <Icon
                icon="gpsArrow"
                title="icon"
                viewBox="0 0 51.636 51.636"
              />
              <span>
                Geolokalizacja
              </span>
            </button>,
          ],
          "className": "Search",
          "style": Object {
            "opacity": "1",
            "transform": "translateY(0)",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="input"
                  onChange={[MockFunction]}
                  placeholder="Szukaj lokalizacji"
                  type="text"
                  value=""
                />,
                <button
                  className="button formButton"
                  type="submit"
                >
                  <Icon
                    icon="search"
                    title="icon"
                    viewBox="0 0 50 50"
                  />
                  <span>
                    Szukaj
                  </span>
                </button>,
              ],
              "className": "form",
              "onSubmit": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input",
                  "onChange": [MockFunction],
                  "placeholder": "Szukaj lokalizacji",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Icon
                      icon="search"
                      title="icon"
                      viewBox="0 0 50 50"
                    />,
                    <span>
                      Szukaj
                    </span>,
                  ],
                  "className": "button formButton",
                  "type": "submit",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "icon": "search",
                      "title": "icon",
                      "viewBox": "0 0 50 50",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Szukaj",
                    },
                    "ref": null,
                    "rendered": "Szukaj",
                    "type": "span",
                  },
                ],
                "type": "button",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Icon
                  icon="gpsArrow"
                  title="icon"
                  viewBox="0 0 51.636 51.636"
                />,
                <span>
                  Geolokalizacja
                </span>,
              ],
              "className": "button buttonGeo",
              "onClick": [MockFunction],
              "type": "button",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "icon": "gpsArrow",
                  "title": "icon",
                  "viewBox": "0 0 51.636 51.636",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Geolokalizacja",
                },
                "ref": null,
                "rendered": "Geolokalizacja",
                "type": "span",
              },
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Backdrop
            hideBackdrop={[MockFunction]}
            showBackdrop={true}
          />,
          <div
            className="Search"
            style={
              Object {
                "opacity": "1",
                "transform": "translateY(0)",
              }
            }
          >
            <form
              className="form"
              onSubmit={[MockFunction]}
            >
              <input
                className="input"
                onChange={[MockFunction]}
                placeholder="Szukaj lokalizacji"
                type="text"
                value=""
              />
              <button
                className="button formButton"
                type="submit"
              >
                <Icon
                  icon="search"
                  title="icon"
                  viewBox="0 0 50 50"
                />
                <span>
                  Szukaj
                </span>
              </button>
            </form>
            <button
              className="button buttonGeo"
              onClick={[MockFunction]}
              type="button"
            >
              <Icon
                icon="gpsArrow"
                title="icon"
                viewBox="0 0 51.636 51.636"
              />
              <span>
                Geolokalizacja
              </span>
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "hideBackdrop": [MockFunction],
            "showBackdrop": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <form
                className="form"
                onSubmit={[MockFunction]}
              >
                <input
                  className="input"
                  onChange={[MockFunction]}
                  placeholder="Szukaj lokalizacji"
                  type="text"
                  value=""
                />
                <button
                  className="button formButton"
                  type="submit"
                >
                  <Icon
                    icon="search"
                    title="icon"
                    viewBox="0 0 50 50"
                  />
                  <span>
                    Szukaj
                  </span>
                </button>
              </form>,
              <button
                className="button buttonGeo"
                onClick={[MockFunction]}
                type="button"
              >
                <Icon
                  icon="gpsArrow"
                  title="icon"
                  viewBox="0 0 51.636 51.636"
                />
                <span>
                  Geolokalizacja
                </span>
              </button>,
            ],
            "className": "Search",
            "style": Object {
              "opacity": "1",
              "transform": "translateY(0)",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="input"
                    onChange={[MockFunction]}
                    placeholder="Szukaj lokalizacji"
                    type="text"
                    value=""
                  />,
                  <button
                    className="button formButton"
                    type="submit"
                  >
                    <Icon
                      icon="search"
                      title="icon"
                      viewBox="0 0 50 50"
                    />
                    <span>
                      Szukaj
                    </span>
                  </button>,
                ],
                "className": "form",
                "onSubmit": [MockFunction],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "input",
                    "onChange": [MockFunction],
                    "placeholder": "Szukaj lokalizacji",
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Icon
                        icon="search"
                        title="icon"
                        viewBox="0 0 50 50"
                      />,
                      <span>
                        Szukaj
                      </span>,
                    ],
                    "className": "button formButton",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "icon": "search",
                        "title": "icon",
                        "viewBox": "0 0 50 50",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Szukaj",
                      },
                      "ref": null,
                      "rendered": "Szukaj",
                      "type": "span",
                    },
                  ],
                  "type": "button",
                },
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Icon
                    icon="gpsArrow"
                    title="icon"
                    viewBox="0 0 51.636 51.636"
                  />,
                  <span>
                    Geolokalizacja
                  </span>,
                ],
                "className": "button buttonGeo",
                "onClick": [MockFunction],
                "type": "button",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "gpsArrow",
                    "title": "icon",
                    "viewBox": "0 0 51.636 51.636",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Geolokalizacja",
                  },
                  "ref": null,
                  "rendered": "Geolokalizacja",
                  "type": "span",
                },
              ],
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
